tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs

```diff
@@ -19,7 +19,18 @@ module AnonRecd =
         |> shouldFail
         |> withErrorCode 3522
         |> withMessage "The field 'A' appears multiple times in this record expression."
-
+    
+    [<Fact>]
+    let ``Anonymous Records incomplete`` () =
+        Fsx """
+let x () : {| A: int; B: string  |} =  {| A = 123 |}
+"""
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 1, Line 2, Col 40, Line 2, Col 53, "This anonymous record does not have enough fields. Add the missing fields [B].")
```

Comment:
Looks like this error message isn't part of this PR, but I think it could also be improved. `enough` seems wrong here, because just having enough fields wouldn't necessarily help if they aren't the correct ones. It should rather be `... doesn't have all <expected|required|defined|specified> fields ...`. Or even just `This anonymous record missing field B` / `This anonymous record missing fields [B, C]`

Not critical though 😄

Reply:
I can do in a separate PR 🙂